cql3/expr/expression.hh

```diff
@@ -65,9 +74,49 @@ class conjunction;
 struct column_value;
 struct column_value_tuple;
 struct token;
+class unresolved_identifier;
+class column_mutation_attribute;
+class function_call;
+class cast;
+class field_selection;
 
 /// A restriction expression -- union of all possible restriction types.  bool means a Boolean constant.
-using expression = std::variant<bool, conjunction, binary_operator, column_value, column_value_tuple, token>;
+using expression = std::variant<bool, conjunction, binary_operator, column_value, column_value_tuple, token,
+                                unresolved_identifier, column_mutation_attribute, function_call, cast,
+                                field_selection>;
+
+template <typename T, typename Variant = expression>
```

Comment:
The template accepts a variant, and the default is the variant that is `expression`. However, when processing, it's instantiated for other variants.

I'll probably divorce with from expressions and make it a utility, not because I expect reuse, but because readers won't suffer from mixing of abstraction levels.

Reply:
Update: I'll divorce it from existing.